scripts/util.py

```diff
@@ -36,34 +38,47 @@ def templates_root():
     return os.path.dirname(script_dir())
 
 ################################################################
-# Read configuration information from the standard input
+# Ask the user for set up information
 
-def read_from_stdin():
-    return input().strip()
+def ask_for_project_name():
+    """Ask user for project name."""
 
-def read_path_from_stdin(description):
-    print("What is the path to {}? ".format(description), end="")
-    return os.path.abspath(os.path.expanduser(read_from_stdin()))
+    return input("What is the project name? ").strip()
 
-def read_source_root_path():
-    return read_path_from_stdin("the source root")
+def ask_for_function_name():
+    """Ask user for function name."""
 
-def read_proof_root_path():
-    return read_path_from_stdin("the 'proofs' directory (usually '.')")
+    return input("What is the function name? ").strip()
 
-def read_litani_path():
-    return read_path_from_stdin("the litani script")
+def ask_for_source_file(func, cwd=None, repo=None):
+    """Ask user to select path to source file defining function func."""
 
-def read_source_path():
-    return read_path_from_stdin("the source file defining the function")
+    cwd = Path(cwd or Path.cwd()).resolve()
+    repo = Path(repo or repository.repository_root(cwd=cwd)).resolve()
+    sources = repository.function_sources(func, cwd=cwd, repo=repo, abspath=False)
+    options = sources + ["The source file is not listed here"]
+    choices = [str(idx) for idx in range(len(options))]
 
-def read_function_name():
-    print("What is the function name? ", end="")
-    return read_from_stdin()
+    index = "0"
+    if sources:
+        print(f"These source files define a function '{func}':")
+        for index, source in enumerate(options):
+            print(f" {index:3} {source}")
+        index = input(
+            f"Select a source file (the options are {', '.join(choices)}): "
+        ).strip()
 
-def read_project_name():
-    print("What is the project name? ", end="")
-    return read_from_stdin()
+    if not index in choices:
```

Comment:
In the example that you provided in the PR's description, `choices` is equal to: `['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11']`

If a user were to enter `12` then `not index in choices` evaluates to `True` and the UserWarning would be raised (as it should).

Both `index not in choices` and `not index in choices` result into the same value. I tend to prefer the first in terms of readability.

Reply:
Good catch.  I've changed both to 'not in'.